common/persistence/sql/sqlQueue.go

```diff
@@ -88,3 +88,52 @@ func (q *sqlQueue) DequeueMessages(lastMessageID, maxCount int) ([]*persistence.
 func newQueueRow(queueType int, messageID int, payload []byte) *sqldb.QueueRow {
 	return &sqldb.QueueRow{QueueType: queueType, MessageID: messageID, MessagePayload: payload}
 }
+
+func (q *sqlQueue) DeleteMessagesBefore(messageID int) error {
+	_, err := q.db.DeleteMessagesBefore(q.queueType, messageID)
+	if err != nil {
+		return &workflow.InternalServiceError{
+			Message: fmt.Sprintf("DeleteMessagesBefore operation failed. Error %v", err),
+		}
+	}
+	return nil
+}
+
+func (q *sqlQueue) UpdateAckLevel(messageID int, clusterName string) error {
+	err := q.txExecute("UpdateAckLevel", func(tx sqldb.Tx) error {
+		clusterAckLevels, err := tx.GetAckLevels(q.queueType, true)
+		if err != nil {
+			return &workflow.InternalServiceError{
+				Message: fmt.Sprintf("UpdateAckLevel operation failed. Error %v", err),
+			}
+		}
+
+		if clusterAckLevels == nil {
+			err := tx.InsertAckLevel(q.queueType, messageID, clusterName)
+			if err != nil {
+				return &workflow.InternalServiceError{
+					Message: fmt.Sprintf("UpdateAckLevel operation failed. Error %v", err),
+				}
+			}
+			return nil
+		}
+
+		clusterAckLevels[clusterName] = messageID
```

Comment:
same reason (plz see cassandra section), only update is message ID is larger than that in DB